packages/synapse-react-client/src/components/SqlDefinedTableEditor/SqlDefinedTableEditorModal.test.tsx

```diff
@@ -54,15 +56,16 @@ describe('SqlDefinedTableEditorModal tests', () => {
     expect(within(dialog).getByRole('button', { name: 'Save' })).toBeVisible()
   })
 
-  it('loads and displays SQL from the given entityId', () => {
+  it('loads and displays SQL from the given entityId', async () => {
     const textarea = screen.getByRole('textbox', { name: 'Defining SQL' })
+    await waitFor(() => expect(textarea).toHaveValue(originalSql))
     expect(textarea).toBeInTheDocument()
```

Comment:
If you used `getByRole`, then it's in the document, so this is an unnecessary assertion
```suggestion
```